game-pathfinderwrathoftherighteous/index.js

```diff
@@ -1,46 +1,134 @@
+      queryPath: findGame,
+      queryModPath: () => 'Mods',
+      executable: () => ExeName,
+      requiredFiles: [ExeName],
+      environment: {
+        SteamAPPId: SteamId,
+      }, 
+      details:
+      {
+        steamAppId: SteamId,
+      },
+      setup: setup,
+    });
+  context.registerInstaller(NexusId + '-mod', 25, testMod, installMod);
+
+  function findGame() {
+    return util.steam.findByAppId(SteamId)
+      .then(game => game.gamePath)
+      .catch(() => readRegistryKey('HKEY_LOCAL_MACHINE',
+        `SOFTWARE\\WOW6432Node\\GOG.com\\Games\\${GogId}`,
+        'PATH'))
+      .catch(() => readRegistryKey('HKEY_LOCAL_MACHINE',
+        `SOFTWARE\\GOG.com\\Games\\${GogId}`,
+        'PATH'))
+  }
+
+  function readRegistryKey(hive, key, name) {
+    if (!IsWin) {
+      return Promise.reject(new util.UnsupportedOperatingSystem());
+    }
+
+    try {
+      const instPath = winapi.RegGetValue(hive, key, name);
+      if (!instPath) {
+        throw new Error('empty registry key');
+      }
+      return Promise.resolve(instPath.value);
+    } catch (err) {
+      return Promise.reject(new util.ProcessCanceled(err));
+    }
+  }
+
+  function findUnityModManager() {
+    return readRegistryKey('HKEY_CURRENT_USER', 'Software\\UnityModManager', 'Path')
+      .then(value => fs.statAsync(path.join(value, ummDll)));
+  }
+
+  function setup(discovery) {
+    return fs.ensureDirWritableAsync(path.join(discovery.path, 'Mods'), () => Promise.resolve())
+      .then(() => findUnityModManager()
+        .catch(err => {
+          return new Promise((resolve, reject) => {
+            context.api.store.dispatch(
+              actions.showDialog(
+                'question',
+                'Action required',
+                { message: 'You must install Unity Mod Manager to use mods with ' + Name + '.' },
+                [
+                  { label: 'Cancel', action: () => reject(new util.UserCanceled()) },
+                  {
+                    label: 'Go to the Unity Mod Manager page', action: () => {
+                      util.opn('https://www.nexusmods.com/site/mods/21/').catch(err => undefined);
+                      reject(new util.UserCanceled());
+                    }
+                  }
+                ]
+              )
+            );
+          });
+        }))
+  }
+
+  function installMod(files, destinationPath) {
+    const infoFile = files.find(file => file.endsWith(ummModInfo));
+    const idx = infoFile.indexOf(ummModInfo);
+    const rootPath = path.dirname(infoFile);
+    const modName = path.basename(destinationPath, '.installing')
+      .replace(/[^A-Za-z]/g, '');
+  
+    const filtered = files.filter(file => (!file.endsWith(path.sep))
+      && (file.indexOf(rootPath) !== -1));
+  
+    const instructions = filtered.map(file => {
+      return {
+        type: 'copy',
+        source: file,
+        destination: path.join(modName, file.substr(idx)),
+      };
     });
-    context.once(() => {
```

Comment:
Pathfinder WotR is designed specifically to work with an enhancement we made to the UMM modtype extension which is responsible of automatically downloading and installing UMM on the user's machine.

Why did you feel the need to remove the `context.api.ext.ummAddGame` call ?
If the UMM extension doesn't work correctly, that's something we should fix, not bypass it by requiring the user to manually install it.